src/app/clusters/basic/basic.cpp

```diff
@@ -62,41 +62,251 @@ CHIP_ERROR BasicAttrAccess::Read(const ConcreteReadAttributePath & aPath, Attrib
+        {
+            productUrl[0] = '\0';
+            status        = CHIP_NO_ERROR;
+        }
+
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(productUrl, strnlen(productUrl, kMaxLen)));
+        }
+        break;
+    }
+
+    case ProductLabel::Id: {
+        constexpr size_t kMaxLen       = DeviceLayer::ConfigurationManager::kMaxProductLabelLength;
+        char productLabel[kMaxLen + 1] = { 0 };
+        status                         = ConfigurationMgr().GetProductLabel(productLabel, sizeof(productLabel));
+
+        // TODO: Remove defaulting once proper runtime defaulting of unimplemented factory data is done
+        if (status == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND || status == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+        {
+            productLabel[0] = '\0';
+            status          = CHIP_NO_ERROR;
+        }
+
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(productLabel, strnlen(productLabel, kMaxLen)));
+        }
+        break;
+    }
+
+    case SerialNumber::Id: {
+        constexpr size_t kMaxLen             = DeviceLayer::ConfigurationManager::kMaxSerialNumberLength;
+        char serialNumberString[kMaxLen + 1] = { 0 };
+        status                               = ConfigurationMgr().GetSerialNumber(serialNumberString, sizeof(serialNumberString));
+
+        // TODO: Remove defaulting once proper runtime defaulting of unimplemented factory data is done
+        if (status == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND || status == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+        {
+            serialNumberString[0] = '\0';
+            status                = CHIP_NO_ERROR;
+        }
+
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(serialNumberString, strnlen(serialNumberString, kMaxLen)));
+        }
+        break;
+    }
+
+    case UniqueID::Id: {
+        constexpr size_t kMaxLen   = DeviceLayer::ConfigurationManager::kMaxUniqueIDLength;
+        char uniqueId[kMaxLen + 1] = { 0 };
+        status                     = ConfigurationMgr().GetUniqueId(uniqueId, sizeof(uniqueId));
+
+        // TODO: Remove defaulting once proper runtime defaulting of unimplemented factory data is done
+        if (status == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND || status == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
         {
-            err = aEncoder.Encode(chip::CharSpan("XX", strlen("XX")));
+            uniqueId[0] = '\0';
+            status      = CHIP_NO_ERROR;
         }
-        else
+
+        if (status == CHIP_NO_ERROR)
         {
-            err = aEncoder.Encode(chip::CharSpan(location, strlen(location)));
+            status = aEncoder.Encode(chip::CharSpan(uniqueId, strnlen(uniqueId, kMaxLen)));
         }
+        break;
+    }
+
+    default:
+        // We did not find a processing path, the caller will delegate elsewhere.
+        status = CHIP_ERROR_NOT_FOUND;
```

Comment:
Why not just set status to `CHIP_NO_ERROR`?

Reply:
Fixed

Reply:
Fixed